Mixer container crashes with 'invalid memory address or nil pointer dereference' after upgrading to 1.1.6

**Bug description**
After upgrading istio from 1.0.6 to 1.1.6 mixer is crashing in both `istio-policy` and `istio-telemetry` pods.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1784e3f]

goroutine 1 [running]:
istio.io/istio/mixer/pkg/runtime/routing.(*builder).addRuleOperations(0xc420ec74c0, 0xc4203412c0, 0xc, 0x0, 0x0, 0xc42000dcb0, 0x1, 0x1)
	/workspace/go/src/istio.io/istio/mixer/pkg/runtime/routing/builder.go:728 +0x12f
istio.io/istio/mixer/pkg/runtime/routing.(*builder).build(0xc420ec74c0, 0xc420036000)
	/workspace/go/src/istio.io/istio/mixer/pkg/runtime/routing/builder.go:234 +0x1011
istio.io/istio/mixer/pkg/runtime/routing.BuildTable(0xc420af9b00, 0xc420036000, 0x7ffecb65593b, 0xc, 0xc420c0f500, 0x4625be)
	/workspace/go/src/istio.io/istio/mixer/pkg/runtime/routing/builder.go:137 +0x286
istio.io/istio/mixer/pkg/runtime.(*Runtime).processNewConfig(0xc42027a070)
	/workspace/go/src/istio.io/istio/mixer/pkg/runtime/runtime.go:159 +0x10b
istio.io/istio/mixer/pkg/runtime.(*Runtime).StartListening(0xc42027a070, 0x0, 0x0)
	/workspace/go/src/istio.io/istio/mixer/pkg/runtime/runtime.go:117 +0x15f
istio.io/istio/mixer/pkg/server.newPatchTable.func1(0xc42027a070, 0xc420531a70, 0xc420531aa0)
	/workspace/go/src/istio.io/istio/mixer/pkg/server/server.go:105 +0x2b
istio.io/istio/mixer/pkg/server.newServer(0xc420334600, 0xc420534190, 0x6977206465747261, 0xa73250a6874, 0xc42054a480)
	/workspace/go/src/istio.io/istio/mixer/pkg/server/server.go:219 +0x926
istio.io/istio/mixer/pkg/server.New(0xc420334600, 0x15, 0xc42054a480, 0x1)
	/workspace/go/src/istio.io/istio/mixer/pkg/server/server.go:95 +0x39
istio.io/istio/mixer/cmd/mixs/cmd.runServer(0xc420334600, 0x20f2c60, 0x20f2c58)
	/workspace/go/src/istio.io/istio/mixer/cmd/mixs/cmd/server.go:93 +0x9e
istio.io/istio/mixer/cmd/mixs/cmd.serverCmd.func1(0xc420279680, 0xc420037980, 0x0, 0x8)
	/workspace/go/src/istio.io/istio/mixer/cmd/mixs/cmd/server.go:37 +0x3c
istio.io/istio/vendor/github.com/spf13/cobra.(*Command).execute(0xc420279680, 0xc420037880, 0x8, 0x8, 0xc420279680, 0xc420037880)
	/workspace/go/src/istio.io/istio/vendor/github.com/spf13/cobra/command.go:760 +0x2c1
istio.io/istio/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420279400, 0x0, 0x2031de8, 0x12)
	/workspace/go/src/istio.io/istio/vendor/github.com/spf13/cobra/command.go:846 +0x30a
istio.io/istio/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420279400, 0x9, 0x9)
	/workspace/go/src/istio.io/istio/vendor/github.com/spf13/cobra/command.go:794 +0x2b
main.main()
	/workspace/go/src/istio.io/istio/mixer/cmd/mixs/main.go:39 +0x293
```

**Expected behavior**
Mixer was able to start up.

**Steps to reproduce the bug**
Currently unclear.

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
Istio is version 1.1.6 inside the cluster, but as mixer is not running --remote will not report it:
```
$ istioctl version --remote
version.BuildInfo{Version:"1.1.4", GitRevision:"bc7465793cbff4c4189639b3f404e21c517cbdc6-dirty", User:"brew", Host:"Mojave-2.local", GolangVersion:"go1.12.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.4"}
Error: error execing into istio-policy-65d47566cd-zfrzb mixer container: unable to upgrade connection: container not found ("mixer")
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-19T22:12:47Z", GoVersion:"go1.12.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.6-eks-d69f1b", GitCommit:"d69f1bf3669bf00b7f4a758e978e0e7a1e3a68f7", GitTreeState:"clean", BuildDate:"2019-02-28T20:26:10Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
1.0.6 was installed using helm with helmsman describing the desired state.
Upgrade was also done using helm, by following the upgrade steps for helm.

**Environment where bug was observed (cloud vendor, OS, etc)**
AWS EKS, worker nodes running kubernetes v1.12.7

Mixer and pilot logs:
[mixer_mixer.log](https://github.com/istio/istio/files/3196670/mixer_mixer.log)
[pilot_discovery.log](https://github.com/istio/istio/files/3196671/pilot_discovery.log)
[pilot_istio_proxy.log](https://github.com/istio/istio/files/3196672/pilot_istio_proxy.log)
 
replicaset definitions:

[replicaset_policy.yml.txt](https://github.com/istio/istio/files/3196673/replicaset_policy.yml.txt)
[replicaset_telemetry.yml.txt](https://github.com/istio/istio/files/3196674/replicaset_telemetry.yml.txt)

We've solved the problem for now. The problem was due to a config.istio.io/v1alpha2/rule existing that pointed to a handler instance that did not exist. After deleting the rule, and waiting for the broken policy and telemetry pods to restart, mixer was able to start up without incidents.

An error was logged in mixer, but execution continued regardless and crashed at a later point, which threw us of for a little while.

```
2019-05-20T11:33:54.458119Z	info	mcp	  [30] istio/config/v1alpha2/templates
2019-05-20T11:33:54.458122Z	info	mcp	  [31] istio/config/v1alpha2/legacy/tracespans
2019-05-20T11:33:54.458125Z	info	mcp	  [32] istio/config/v1alpha2/legacy/zipkins
2019-05-20T11:33:54.458129Z	info	mcp	  [33] istio/policy/v1beta1/attributemanifests
2019-05-20T11:33:54.458132Z	info	mcp	  [34] istio/config/v1alpha2/legacy/deniers
2019-05-20T11:33:54.458135Z	info	mcp	  [35] istio/policy/v1beta1/instances
2019-05-20T11:33:54.458138Z	info	mcp	  [36] istio/config/v1alpha2/legacy/statsds
2019-05-20T11:33:54.458152Z	info	parsed scheme: ""
2019-05-20T11:33:54.458160Z	info	scheme "" not registered, fallback to default scheme
2019-05-20T11:33:54.458243Z	info	Using new MCP client sink stack
2019-05-20T11:33:54.458293Z	info	Awaiting for config store sync...
2019-05-20T11:33:54.458482Z	info	mcp	(re)trying to establish new MCP sink stream
2019-05-20T11:33:54.458615Z	info	ccResolverWrapper: sending new addresses to cc: [{istio-galley.istio-system.svc:9901 0  <nil>}]
2019-05-20T11:33:54.458624Z	info	ClientConn switching balancer to "pick_first"
2019-05-20T11:33:54.458663Z	info	pickfirstBalancer: HandleSubConnStateChange: 0xc4204d2250, CONNECTING
2019-05-20T11:33:54.467799Z	info	pickfirstBalancer: HandleSubConnStateChange: 0xc4204d2250, READY
2019-05-20T11:33:54.467885Z	info	mcp	New MCP sink stream created
2019-05-20T11:33:54.488391Z	info	Starting runtime config watch...
2019-05-20T11:33:54.488462Z	info	Built new config.Snapshot: id='0'
2019-05-20T11:33:54.488476Z	info	Cleaning up handler table, with config ID:-1
2019-05-20T11:33:54.515095Z	error	Handler not found: handler='keyval.istio-system'
gc 3 @0.092s 7%: 0.041+22+0.062 ms clock, 0.082+5.5/21/7.9+0.12 ms cpu, 6->7->5 MB, 7 MB goal, 6 P
2019-05-20T11:33:54.516630Z	error	Handler not found: handler='keyval.istio-system'
2019-05-20T11:33:54.516642Z	info	Built new config.Snapshot: id='1'
2019-05-20T11:33:54.517067Z	info	adapters	getting kubeconfig from: ""	{"adapter": "kubernetesenv.istio-system"}
2019-05-20T11:33:54.517078Z	warn	Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2019-05-20T11:33:54.517906Z	info	adapters	Waiting for kubernetes cache sync...	{"adapter": "kubernetesenv.istio-system"}
2019-05-20T11:33:54.518183Z	info	adapters	serving prometheus metrics on 42422	{"adapter": "prometheus.istio-system"}
gc 4 @0.151s 7%: 0.031+23+2.8 ms clock, 0.063+1.3/17/0.51+5.6 ms cpu, 9->11->7 MB, 10 MB goal, 6 P
2019-05-20T11:33:54.618040Z	info	adapters	Cache sync successful.	{"adapter": "kubernetesenv.istio-system"}
2019-05-20T11:33:54.618080Z	info	adapters	getting kubeconfig from: ""	{"adapter": "kubernetesenv.istio-system"}
2019-05-20T11:33:54.618086Z	warn	Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2019-05-20T11:33:54.618867Z	info	Setting up event handlers
2019-05-20T11:33:54.618967Z	info	Starting Secrets controller
2019-05-20T11:33:54.618990Z	info	Waiting for informer caches to sync
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1784e3f]
```

--- a/mixer/pkg/runtime/routing/builder.go
+++ b/mixer/pkg/runtime/routing/builder.go
@@ -734,15 +734,17 @@ func (b *builder) addRuleOperations(
 		b.table.entries[tpb.TEMPLATE_VARIETY_CHECK] = &varietyTable{}
 	}
 	if b.table.entries[tpb.TEMPLATE_VARIETY_CHECK].entries == nil {
-		b.table.entries[tpb.TEMPLATE_VARIETY_CHECK].entries = map[string]*NamespaceTable{
-			namespace: {
-				entries:    []*Destination{},
-				directives: []*DirectiveGroup{},
-			},
-		}
+		b.table.entries[tpb.TEMPLATE_VARIETY_CHECK].entries = make(map[string]*NamespaceTable)
 	}
 
-	byNamespace := b.table.entries[tpb.TEMPLATE_VARIETY_CHECK].entries[namespace]
+	byNamespace, found := b.table.entries[tpb.TEMPLATE_VARIETY_CHECK].entries[namespace]
+	if !found {
+		byNamespace = &NamespaceTable{
+			entries:    []*Destination{},
+			directives: []*DirectiveGroup{},
+		}
+		b.table.entries[tpb.TEMPLATE_VARIETY_CHECK].entries[namespace] = byNamespace
+	}
 
 	var group *DirectiveGroup
 	for _, set := range byNamespace.directives {

--- a/mixer/pkg/runtime/routing/builder_test.go
+++ b/mixer/pkg/runtime/routing/builder_test.go
@@ -21,6 +21,7 @@ import (
 
 	"github.com/gogo/protobuf/types"
 
+	tpb "istio.io/api/mixer/adapter/model/v1beta1"
 	"istio.io/istio/mixer/pkg/adapter"
 	"istio.io/istio/mixer/pkg/runtime/config"
 	"istio.io/istio/mixer/pkg/runtime/handler"
@@ -563,6 +564,30 @@ func buildTableWithTemplatesAndAdapters(templates map[string]*template.Info, ada
 	ht := handler.NewTable(handler.Empty(), s, nil)
 
 	return BuildTable(ht, s, "istio-system", debugInfo), s
+}
+
+func TestAddRuleOperations(t *testing.T) {
+	b := &builder{
+		table: &Table{entries: make(map[tpb.TemplateVariety]*varietyTable)},
+	}
+	// put something into the table for a different namespace
+	b.table.entries[tpb.TEMPLATE_VARIETY_CHECK] = &varietyTable{
+		entries: map[string]*NamespaceTable{
+			"ns2": {
+				entries:    []*Destination{},
+				directives: []*DirectiveGroup{},
+			},
+		},
+	}
+
+	// catch the panic
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fail()
+		}
+	}()
+
+	b.addRuleOperations("ns1", nil, nil)
 }
 
 func TestNonPointerAdapter(t *testing.T) {

